### PR TITLE
bpo-32117: Allow tuple unpacking in return and yield statements

### PR DESCRIPTION
**Pull request by dacut**
This is a fix for [issue 32117](https://bugs.python.org/issue32117):

This stems from a [query on StackOverflow](https://stackoverflow.com/questions/47272460/python-tuple-unpacking-in-return-statement/).

Specifically, the following syntax is allowed:

    def f():
        rest = (4, 5, 6)
        t = 1, 2, 3, *rest

While the following result in SyntaxError:

    def g():
        rest = (4, 5, 6)
        return 1, 2, 3, *rest

    def h():
        rest = (4, 5, 6)
        yield 1, 2, 3, *rest

Looking at the original commit that enabled tuple unpacking in assignment statements:
https://github.com/python/cpython/commit/4905e80c3d2f6abb613d212f0313d1dfe09475dc

I don't believe this difference is intentional.

https://bugs.python.org/issue32117

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by dacut** [comment]
CLA signed and awaiting review.

**Comment by dacut** [comment]
CLA reviewed and approved.

**Comment by gvanrossum** [comment]
I'm going to close and reopen to cause Travis-CI and AppVeyor to rerun the tests.

File: Lib/test/test_grammar.py
Comment (gvanrossum) [inline Lib/test/test_grammar.py:991]: I would add another test here to validate that this works, e.g. `self.assertEqual(list(g()), [(1, 2, 3, 4, 5, 6)])`.

**Review by gvanrossum — COMMENTED** [review]
Assuming @dacut is no longer interested in this PR. Jordan Chapman, can you add the test I requested and rebase, then push?

**Comment by dacut** [comment]
I am. :-) I'll add that test if Jordan doesn't beat me to it.

**Comment by jChapman** [comment]
I've added the test to my fork (https://github.com/jChapman/cpython) and rebased, but I don't have the ability to modify this pull request, so I am unsure how to proceed.

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by dacut** [comment]
Gah, that is not what I had intended. Reverting... Give me a sec.

**Comment by dacut** [comment]
I've pulled @jChapman's repo and pushed it to mine to update this request.
Sometimes `git push --force` *is* what you want. :-)

**Comment by jChapman** [comment]
@dacut probably because I just signed today and now the PR contains one of my commits (sorry)

File: Lib/test/test_grammar.py
Comment (jChapman) [reply Lib/test/test_grammar.py:991]: I was tempted to do `self.assertEqual(*g(), (1, 2, 3, 4, 5, 6))`, but I think we're testing enough unpacking

**Review by gvanrossum — APPROVED** [review]
This looks good to me. The code generation doesn't need any changes because `return (a, b, *x)` and `return a, b, *x` produce the same AST. I'll see if I can fix the redundant review requests.

**Comment by gvanrossum** [comment]
Also, I checked that CLA from @jChapman was received so I'll change those labels. I'll probably wait a day before merging, in case another core dev wants to have a look and/or say.

**Comment by jChapman** [comment]
@gvanrossum Reminder to complete merge (it has now been 4 days)

**Comment by brettcannon** [comment]
[Merge unpacking return value support](https://trello.com/c/EDngiFfJ/131-merge-unpacking-return-value-support)

**Comment by gvanrossum** [comment]
@brettcannon What's that trello.com link? I get "Card not found".

**Comment by brettcannon** [comment]
I touched up the news entry slightly (made sure to thank @dacut and @jChapman 😄 ). If @gvanrossum doesn't merge this by the time I do my next sweep through my Python TODO list then I will.

**Comment by brettcannon** [comment]
@gvanrossum the Trello card is for my private board to keep track of my TODO entries; didn't realize it left a comment. :/

**Comment by gvanrossum** [comment]
I see no reason to hold this up. Thanks @dacut for the fix and @jChapman for pinging this!

**Comment by gvanrossum** [comment]
Well, I think the news entry should say "parentheses" instead of "parenthesis", and we need an entry in whatsnew.rst, but that can be done in a followup PR (@dacut and @jChapman, if one of you would be so kind?).

Also it's ridiculous what GitHub did to my commit message, but too late to fix now.

**Pull request merged**
